src/System.Collections/tests/Performance/Perf.Dictionary.cs

```diff
@@ -170,28 +170,339 @@ public void TryGetValue(int size)
         }
 
         [Benchmark]
+        [InlineData(1)]
+        [InlineData(10)]
         [InlineData(1000)]
-        [InlineData(10000)]
-        [InlineData(100000)]
-        public void ContainsKey(int size)
+        public static void Remove_ValueType(long size)
         {
-            Dictionary<int, int> dict = CreateDictionary(size);
+            Dictionary<long?, long?> collection = new Dictionary<long?, long?>();
+            long?[] items;
+
+            items = new long?[size * 10];
+
+            for (long i = 0; i < size * 10; ++i)
+            {
+                items[i] = i;
+                collection.Add(items[i], items[i]);
+            }
 
-            // Setup - utils needs a specific seed to prevent key collision with TestData
-            Random rand = new Random(837322);
-            int key = rand.Next(0, 400000);
-            dict.Add(key, 12);
 
-            // Actual perf testing
             foreach (var iteration in Benchmark.Iterations)
                 using (iteration.StartMeasurement())
-                    for (int i = 0; i <= 10000; i++)
+                    for (long i = 1; i < size; ++i)
+                        collection.Remove(items[i]);
+        }
+
+        [Benchmark]
+        public static void Indexer_get_ValueType()
+        {
+            int size = 1024;
+            int? item;
+            Dictionary<int?, int?> collection = new Dictionary<int?, int?>();
+            for (int i = 0; i < size; ++i)
+            {
+                collection.Add(i, i);
+            }
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                {
+                    for (int j = 0; j < size; ++j)
                     {
-                        dict.ContainsKey(key); dict.ContainsKey(key); dict.ContainsKey(key);
-                        dict.ContainsKey(key); dict.ContainsKey(key); dict.ContainsKey(key);
-                        dict.ContainsKey(key); dict.ContainsKey(key); dict.ContainsKey(key);
-                        dict.ContainsKey(key); dict.ContainsKey(key); dict.ContainsKey(key);
+                        item = (int)collection[j];
```

Comment:
Not sure... I'm wondering if it works around a dead store optimization in the JIT.  Otherwise, I would expect this loop to only do one fetch and store.